pyiron_base/generic/hdfio.py

```diff
@@ -43,8 +45,11 @@ def _is_ragged_array(value: Union[np.ndarray, list]) -> bool:
     if isinstance(value, np.ndarray) and value.dtype != np.dtype("O"):
         return False
     else:
-        shape_lst = [np.shape(sub) for sub in value]
-        return len(set(shape_lst)) > 1
+        def extract_dims(v):
+            s = np.shape(v)
```

Comment:
For this call, I already get the numpy deprecation warning... Do we want to catch that or what to do if they really take the automated conversion to dtype=object out?...